diff --git a/build/browser.d.ts b/build/browser.d.ts
index 8a6738a7aa0c2d8078536fea0f79654a5adfb36c..763b44648d5ddbdddd5d6899d5387166645a7e2c 100644
--- a/build/browser.d.ts
+++ b/build/browser.d.ts
@@ -4,3 +4,4 @@ export * from './high-level/types';
 export { Z3Core, Z3LowLevel } from './low-level';
 export * from './low-level/types.__GENERATED__';
 export declare function init(): Promise<Z3LowLevel & Z3HighLevel>;
+export declare function initWith(initZ3: any): Promise<Z3LowLevel & Z3HighLevel>;
diff --git a/build/browser.js b/build/browser.js
index 5334ed98f409d42b33fa4e20d0a808647bab2d3a..f222f3d76740beaadf50d0a7515d1f8e01dd4d1d 100644
--- a/build/browser.js
+++ b/build/browser.js
@@ -15,6 +15,7 @@ var __exportStar = (this && this.__exportStar) || function(m, exports) {
 };
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.init = init;
+exports.initWith = initWith;
 const high_level_1 = require("./high-level");
 const low_level_1 = require("./low-level");
 __exportStar(require("./high-level/types"), exports);
@@ -28,3 +29,8 @@ async function init() {
     const highLevel = (0, high_level_1.createApi)(lowLevel.Z3);
     return { ...lowLevel, ...highLevel };
 }
+async function initWith(initZ3) {
+    const lowLevel = await (0, low_level_1.init)(initZ3);
+    const highLevel = (0, high_level_1.createApi)(lowLevel.Z3);
+    return { ...lowLevel, ...highLevel };
+}
diff --git a/build/node.d.ts b/build/node.d.ts
index 503719452e803f5a6e4369d2e5a5cd4ee67b59fd..593a1bb1fafd701ab5f50fc8dcd2ef54525e66cc 100644
--- a/build/node.d.ts
+++ b/build/node.d.ts
@@ -28,3 +28,4 @@ export * from './low-level/types.__GENERATED__';
  * ```
  * @category Global */
 export declare function init(): Promise<Z3HighLevel & Z3LowLevel>;
+export declare function initWith(initZ3: any): Promise<Z3HighLevel & Z3LowLevel>;
diff --git a/build/node.js b/build/node.js
index b83fa11328c4007343f60468d04a43be62689e94..df40d04c981bac3125d71fefaf70d5d2e7161bea 100644
--- a/build/node.js
+++ b/build/node.js
@@ -15,6 +15,7 @@ var __exportStar = (this && this.__exportStar) || function(m, exports) {
 };
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.init = init;
+exports.initWith = initWith;
 // @ts-ignore no-implicit-any
 const initModule = require("./z3-built");
 const high_level_1 = require("./high-level");
@@ -50,3 +51,8 @@ async function init() {
     const highLevel = (0, high_level_1.createApi)(lowLevel.Z3);
     return { ...lowLevel, ...highLevel };
 }
+async function initWith(initZ3) {
+    const lowLevel = await (0, low_level_1.init)(initZ3);
+    const highLevel = (0, high_level_1.createApi)(lowLevel.Z3);
+    return { ...lowLevel, ...highLevel };
+}
